internal/server/routes.go

```diff
@@ -16,114 +16,111 @@ import (
-		get(a, authorized, "/identities/:id/grants", a.ListIdentityGrants)
-
-		get(a, authorized, "/access-keys", a.ListAccessKeys)
-		post(a, authorized, "/access-keys", a.CreateAccessKey)
-		delete(a, authorized, "/access-keys/:id", a.DeleteAccessKey)
-
-		get(a, authorized, "/introspect", a.Introspect)
-
-		get(a, authorized, "/groups", a.ListGroups)
-		post(a, authorized, "/groups", a.CreateGroup)
-		get(a, authorized, "/groups/:id", a.GetGroup)
-		get(a, authorized, "/groups/:id/grants", a.ListGroupGrants)
+	authn := api.Group("/", AuthenticationMiddleware(a))
+	get(a, authn, "/v1/identities", a.ListIdentities)
+	post(a, authn, "/v1/identities", a.CreateIdentity)
+	get(a, authn, "/v1/identities/:id", a.GetIdentity)
+	put(a, authn, "/v1/identities/:id", a.UpdateIdentity)
+	delete(a, authn, "/v1/identities/:id", a.DeleteIdentity)
+	get(a, authn, "/v1/identities/:id/groups", a.ListIdentityGroups)
+	get(a, authn, "/v1/identities/:id/grants", a.ListIdentityGrants)
 
-		get(a, authorized, "/grants", a.ListGrants)
-		get(a, authorized, "/grants/:id", a.GetGrant)
-		post(a, authorized, "/grants", a.CreateGrant)
-		delete(a, authorized, "/grants/:id", a.DeleteGrant)
+	get(a, authn, "/v1/access-keys", a.ListAccessKeys)
+	post(a, authn, "/v1/access-keys", a.CreateAccessKey)
+	delete(a, authn, "/v1/access-keys/:id", a.DeleteAccessKey)
 
-		post(a, authorized, "/providers", a.CreateProvider)
-		put(a, authorized, "/providers/:id", a.UpdateProvider)
-		delete(a, authorized, "/providers/:id", a.DeleteProvider)
+	get(a, authn, "/v1/groups", a.ListGroups)
+	post(a, authn, "/v1/groups", a.CreateGroup)
+	get(a, authn, "/v1/groups/:id", a.GetGroup)
+	get(a, authn, "/v1/groups/:id/grants", a.ListGroupGrants)
 
-		get(a, authorized, "/destinations", a.ListDestinations)
-		get(a, authorized, "/destinations/:id", a.GetDestination)
-		post(a, authorized, "/destinations", a.CreateDestination)
-		put(a, authorized, "/destinations/:id", a.UpdateDestination)
-		delete(a, authorized, "/destinations/:id", a.DeleteDestination)
+	get(a, authn, "/v1/grants", a.ListGrants)
+	get(a, authn, "/v1/grants/:id", a.GetGrant)
+	post(a, authn, "/v1/grants", a.CreateGrant)
+	delete(a, authn, "/v1/grants/:id", a.DeleteGrant)
 
-		post(a, authorized, "/tokens", a.CreateToken)
+	post(a, authn, "/v1/providers", a.CreateProvider)
+	put(a, authn, "/v1/providers/:id", a.UpdateProvider)
+	delete(a, authn, "/v1/providers/:id", a.DeleteProvider)
 
-		post(a, authorized, "/logout", a.Logout)
-	}
+	get(a, authn, "/v1/destinations", a.ListDestinations)
+	get(a, authn, "/v1/destinations/:id", a.GetDestination)
+	post(a, authn, "/v1/destinations", a.CreateDestination)
+	put(a, authn, "/v1/destinations/:id", a.UpdateDestination)
+	delete(a, authn, "/v1/destinations/:id", a.DeleteDestination)
 
-	// these endpoints are left unauthenticated
-	unauthorized := v1.Group("/")
+	get(a, authn, "/v1/introspect", a.Introspect)
+	post(a, authn, "/v1/tokens", a.CreateToken)
+	post(a, authn, "/v1/logout", a.Logout)
```

Comment:
You may notice these 3 moved lines. I moved some of these miscellaneous handlers into a section together, instead of having 3 ungrouped.